rest-api/dao/physical_measurements_dao.py

```diff
@@ -327,6 +344,32 @@ def _update_amended(self, obj, extension, url, session):
     session.merge(amended_measurement)
     obj.amendedMeasurementsId = amended_measurement_id
 
+  def update_with_patch(self, id_, resource):
+    session, measurement = self.get_with_children(id_, for_update=True)
+    return self._do_update_with_patch(session, measurement, resource)
+
+  def _do_update_with_patch(self, session, measurement, resource):
+    self._validate_patch_update(measurement, resource)
+    site_id, author = self._validate_and_get_author_and_site(resource)
+    measurement.reason = resource['reason']  # @TODO: ADD REASON TO DB
+    if resource['status'].lower() == 'cancelled':
+      measurement.cancelledUsername = author
+      measurement.cancelledSiteId = site_id
+      measurement.cancelledTime = clock.CLOCK.now()
+      measurement.status = PhysicalMeasurementsStatus.CANCELLED
+    if resource['status'].lower() == 'restored':
+      measurement.cancelledUsername = None
+      measurement.cancelledSiteId = None
+      measurement.cancelledTime = None
+      measurement.status = PhysicalMeasurementsStatus.UNSET
+
+    payload = self.add_root_fields_to_resource(measurement)
```

Comment:
So site_id, author, and reason all came from the resource to begin with.The only thing that seems like it might need updating is status and cancelledTime?

Reply:
Sorry forgot this was a patch.